src/bson.rs

```diff
@@ -1035,6 +1056,13 @@ pub struct Binary {
     pub bytes: Vec<u8>,
 }
 
+impl Display for Binary {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        let x: Bson = self.into();
```

Comment:
These `.into()` calls will make copies, which can potentially be costly. Instead, can we move the meat of the `Display` implementation here and call into it from `Bson`? This will let us avoid a copy. Ditto for the other types.